python/pyarrow/memory.pxi

```diff
@@ -44,6 +44,15 @@ cdef class LoggingMemoryPool(MemoryPool):
         self.init(self.logging_pool.get())
 
 
+cdef class ProxyMemoryPool(MemoryPool):
```

Comment:
Docstring

Reply:
Other classes are not documented in memory.pxi. Is there another location for documentation I'm missing?

Reply:
No, this is the main location where we should document things. We were not so good with adding docstrings in the past. I'll probably make a pass over the old classes soon and add some documentation but we should start adding documentation to new classes once we add them.

Reply:
Got it. Would be happy to help with the old classes as well.

Reply:
> Would be happy to help with the old classes as well.

That would be very much appreciated!